### ISSUE TITLE
Error when using host:port in a FROM instruction

### ISSUE THREAD

**Issue by ArnaudGallardo** [hadolint/hadolint#355]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

The linter should run on Dockerfiles that use a custom registry with the following format *host:port*.

### Actual behavior

Returns an error before running any linting checks: `unexpected ':' expecting '@', a new line followed by the next instruction, at least one space, or the image tag`.

### Steps to reproduce the behavior

Create the following Dockerfile:

```FROM myregistry:443/imagename:tag```

Run `docker run --rm -i hadolint/hadolint < Dockerfile`.

The error will be triggered.

If you add a `.anything` after `myregistry` or if you remove `:443` it will work. 

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.17.2-8-g65736cb
```

**Dockerfile** (if relevant)

```Dockerfile
FROM myregistry:443/imagename:tag
```

We are using an Artifactory repository on a local network, that's why we have a `host:port` registry.

**Comment by quixoticmonk** [comment]
Noticing the same in one of our cases too with Nexus registries.  Since this is an open issue, I am assuming there is not a fix in the latest versions. Can I add an ignore on this by any means ? Thank you

**Comment by lorenzo** [comment]
There is no workaround, unfortunately, it is a deficiency in the parser. I'll include this fix for next version

**Comment by quixoticmonk** [comment]
Thanks @lorenzo .

**Comment by srcshelton** [comment]
In addition, I have a multi-stage Dockerfile containing:

```
ARG stage3_image="gentoo/stage3-amd64"
ARG stage3_name="gentoo-stage3"

FROM "${stage3_name:-gentoo-stage3}:latest"
```

... which also generates the same error, for character 36 (`:`) of the `FROM` line.

Version `Haskell Dockerfile Linter v1.18.0-6-ga0d655d` run via docker/podman.

**Comment by tombailo** [comment]
Are there any plans to fix this? It is still present in the latest version and blocks me from using hadolint on a project I'm developing
